app/models/service_terraform_template.rb

```diff
@@ -123,15 +167,31 @@ def job_option_key(action)
     "#{action.downcase}_job_options".to_sym
   end
 
-  def parse_dialog_options
-    dialog_options = options[:dialog] || {}
+  # We only want keys starting with 'dialog_', as these are user inputs, don't want other key/value pairs.
+  # Particularly in case of Reconfigure action, we want to remove keys like,
+  # - request => "service_reconfigure"
+  # - Service::service => ##
+  def parse_dialog_options_only(action_options)
+    dialog_options = action_options[:dialog] || {}
+    params = {}
+    dialog_options.each do |attr, val|
+      if attr.start_with?("dialog_")
+        var_key = attr.sub(/^(password::)?dialog_/, '')
+        params[var_key] = val
+      end
+    end
+    params.blank? ? {} : {:input_vars => params}
```

Comment:
Is this necessary?  I don't see how params could become `nil` which looks like the only time `params` would be `blank?=>true` but not `{}`.